Content.Server/Roles/RoleTimerManager.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Content.Server.Database;
+using Content.Shared.Roles;
+using Robust.Server.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Network;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Roles
+{
+    public sealed class RoleTimerManager : IEntityEventSubscriber
+    {
+        [Dependency] private readonly IEntityNetworkManager _netManager = default!;
+        [Dependency] private readonly IServerDbManager _db = default!;
+
+        /*
+         * All of the reads below just read directly from PlayerRoleTimers without touching the DB.
+         * The writes update the timer then write the value back to the db.
+         */
+
+        // Shouldn't need concurrentdictionary as we're only modifying the value internally.
+        private readonly Dictionary<NetUserId, PlayerRoleTimers> _cachedPlayerData = new();
+
+        public async void CreatePlayer(NetUserId id)
+        {
+            var data = _cachedPlayerData.GetOrNew(id);
+            await CachePlayerRoles(id, data);
+        }
+
+        public async void RemovePlayer(NetUserId id)
+        {
+            _cachedPlayerData.Remove(id);
+        }
+
+        public async void SendRoleTimers(IPlayerSession pSession)
+        {
+            var overall = await GetOverallPlaytime(pSession.UserId);
+            var roles = await GetRolePlaytimes(pSession.UserId);
+
+            _netManager.SendSystemNetworkMessage(new RoleTimersEvent()
+            {
+                Overall = overall,
+                RoleTimes = roles,
+            }, pSession.ConnectedClient);
+        }
+
+        public bool IsRoleTimeCachedYet(NetUserId id)
```

Comment:
Returning false if the player doesn't exist feels error prone. That should maybe throw an error instead.